python/ray/train/v2/_internal/execution/worker_group/poll.py

```diff
@@ -31,9 +61,55 @@ def finished(self) -> bool:
         )
 
     def get_error_string(self) -> str:
-        return "\n".join(
-            f"[Rank {world_rank}]\n{error}" for world_rank, error in self.errors.items()
-        )
+        """
+        Returns a string representation worker group errors.
+        Groups similar errors (ignoring numbers) and shows original error examples.
+        """
+        # Group errors by normalized strings (ignoring numbers)
+        normalized_error_to_ranks = defaultdict(list)
+        normalized_error_to_original = {}
+        show_full_error = set()
+
+        for world_rank, status in self.worker_statuses.items():
+            # Exclude errors from running workers
+            if status.error and not status.running:
+                error_str = str(status.error)
+                normalized_error = _normalize_error_string(error_str)
+
+                normalized_error_to_ranks[normalized_error].append(str(world_rank))
+
+                # Store the first original error for this normalized group
+                if normalized_error not in normalized_error_to_original:
+                    normalized_error_to_original[normalized_error] = error_str
+
+                # Fully show errors for non-graceful worker failures
+                if isinstance(status.error, WorkerHealthCheckFailedError):
+                    show_full_error.add(normalized_error)
+
+        # Convert rank lists to comma-separated strings
+        for normalized_error, ranks in normalized_error_to_ranks.items():
+            normalized_error_to_ranks[normalized_error] = ", ".join(ranks)
+
+        errors = []
+        for normalized_error, ranks in normalized_error_to_ranks.items():
+            # Show the original error if there were no duplicates
+            error = (
+                normalized_error
+                if "," in ranks
+                else normalized_error_to_original[normalized_error]
+            )
+
+            if normalized_error in show_full_error:
+                errors.append(f"[Rank {ranks}]:\n{error}")
+            else:
+                errors.append(f"[Rank {ranks}]:\n{_truncate_error_string(error)}")
+
+        error_str = "\n".join(errors)
+
+        if "..." in error_str:
+            error_str += "\nView individual worker logs for more details."
```

Comment:
Rather than doing this once at the end of the entire error message, any reason not to just append this directly to the individual truncated error messages (in `_truncate_error_string`)